### DISCUSSION TITLE
Why do we need batching in signals if React batches state updates?

### DISCUSSION THREAD

**Discussion by crabvk** [preactjs/signals#501] [Q&A] [answered]
Just stumbled upon the [`batch` function in the guide](https://preactjs.com/guide/v10/signals/#batchfn) which led to the question.

**Comment by rschristian** [comment, marked as answer]
React _can_ batch state updates, but there's no guarantees how or when they will. `batch()` provides this guarantee.

```js
import { signal, computed, effect } from '@preact/signals-react';

const fName = signal('John');
const lName = signal('Smith');
const fullName = computed(() => `${fName.value} ${lName.value}`);

effect(() => console.log(fullName.value));

fName.value = 'Jane';
lName.value = 'Doe';
```

This could log:

```
John Smith
Jane Smith
Jane Doe
```

Batching the name updates will always log this:

```
John Smith
Jane Doe
```

**Reply by rschristian** [reply]
Of course, signals also integrate into Preact and are also provided as UI framework-agnostic core library, React's behavior is just a piece of the whole.
